### ISSUE TITLE
Feedback zur Datenbankdatei

### ISSUE THREAD

**Issue by nittka** [nsalomon/TVTower-ariksan-DB#1]
Die für TVTower beste Möglichkeit, Feedback zu den Daten zu geben, wäre ein Fork des offiziellen Repository. Du könntest Deine Daten dann in einem Branch schon an die "richtige Stelle" (User-DB-Daten) legen und einen Pull-Request stellen. Der kann dann kommentiert und, wenn alle Parteien zufrieden sind, direkt übernommen werden.

Die Hinweise, die ich habe, sind zunächst einmal formaler Natur. Damit ich sie Dir gezielt geben kann, werde ich einen PR auf Dein Repository vorbereiten.

**Comment by nsalomon** [comment]
Salü Nittka

Vielen Herzlichen Dank für deine Mühen, das xml.-file durchzuschauen.
Werde alle deine Punkte beherzigen und entsprechend überarbeiten.

Um die Tiere in "Die Sendung mit..." anzugleichen, habe ich, zumindest theoretisch, eine Lösung, die ich asap testen werde.
Die Variablenverwendung wird zwar Aufgrund der unterschiedlchen Grammatik weiterhin nicht 1:1 sein, aber das Ergebnis sollte stimmen.

**Comment by nsalomon** [comment]
```
<!--TODO licenceflags gibt es nicht - production_licence_flags und für die ist Wert 512 ungültig - was war die Intention?, -->
<data flags="9" licenceflags="4" production_broadcast_limit="1" live_date="3,5,20,20" production_licence_flags="512"/>
```

Die Intention für licenceflags="4" war, das die Sendung nur einmal Live ausgestrahlt werden darf und dann verschwindet und keine Wiederholung erlaubt ist.

Dies ist dann wohl mit production_broadcast_limit="1" sooderso überflüssig.

Initial habe ich die Relation zwischen production_licence_flags und licenceflags nicht verstanden.
production_licence_flags im Script wird dann also in der eigentlich produzierten Lizenz zu licence_flags, und wird im Scipt über production_licence_flags definiert.

Was ich mit der 512 wollte bin ich mir nach nochmaliger Durchsicht der Dokumentation ehrlich gesagt gar nicht mehr so sicher.
Glaube ich habe das auf "immer Live" bezogen, was natürlich so nicht gültig / nötig ist.

**Comment by nsalomon** [comment]
So, jetzt weiss ich wieder woher ich das habe....

aus ronny.xml

Zeile 923 -> 923

```
			<!-- live whenever first broadcastd -->
			<!-- licence flags: 4 + 32 (REMOVE_ON_REACHING_BROADCASTLIMIT + LICENCEPOOL_REMOVES_TRADEABILITY) -->
			<data flags="1" broadcast_time_slot_start="5" broadcast_time_slot_end="10" production_limit="4" production_broadcast_limit="1" production_broadcast_flags="512" production_licence_flags="36" />
```

**Comment by nittka** [comment]
Ja, in die Flags muss man sich immer wieder reindenken. Wichtig ist, dass einmalige Ausstrahlung noch nicht automatisch bedeutet, dass die Sendung nicht nochmal gesendet werden kann. Je nach Flags kann sie nach der Ausstrahlung auch zurück zum Filmhändler gehen und dann wieder gekauft und erneut gesendet werden (und sie kann verschwinden ohne dass man den Restwert bekommt oder dass einem dieser gutgeschrieben wird).

**Comment by nittka** [comment]
Für die Sendung mit der Laus würde ich gern nochmal was ausprobieren... Man kann Einzelfolgen auch als optional markieren (episodes min=0 max=1). Wenn man dann 30 mögliche Folgen mit unterschiedlichen Titeln/Beschreibungen definiert, aber im Parent nur 6 erlaubt, bekommt man in jeder Staffel möglicherweise eine andere Folgenzusammestellung und erreicht damit zusätzliche Variabilität.
Es kann aber auch sein, dass für dieses (gewünschte) Verhalten noch was am Code geschraubt werden muss (insb. der Fall dass unwahrscheinlicherweise für jede Einzelfolge 0 als Anzahl gewürfelt wird).

**Comment by nsalomon** [comment]
Jetzt wo ich weiss, wie ich mit einem Element mehrere Folgen generieren kann, werde ich dies evtl. auch für "die Sending mit" entsprechend überarbeiten, so dass jede Episode generiert wird.
Ansonsten werde ich deine Idee umsetzen.

Wo ich nicht ganz durchblicke ist bei "script-ariksan-mkw" linie 30/31

```
				<scripttemplate index="0" product="2" licence_type="2" guid="script-ariksan-mkw01">
					<title>
						<de>%wohnobjekt%</de>
						<en>%wohnobjekt%</en>
					</title>
					<description>
						<de>%wohnfinder%|%wohnsucher%</de>
						<en>%wohnfinder%|%wohnsucher%</en>
					</description>
```

Eigentlich müsste hier random für jede Folge entweder %wohnfinder% oder %wohnsucher% ausgewählt werden. Aber es scheint so, dass hier für alle 8 Folgen immer die gleiche Option gewählt wird. Bei allen anderen Variablen ist es aber kein Problem und jede Folge wird individuel zusammen gestellt, mit Ausnahme dieser Variable.

**Comment by nsalomon** [comment]
Nunja, Problem gelöst mit der 'neuen' Methode. So funktioniert das Problemlos.

```
					<description>
						<de>${wohn${wohnselector}}</de>
						<en>${wohn${wohnselector}}</en>
					</description>
```

**Comment by nittka** [comment]
Ja, sorry. Die Änderung hatte ich ohne Ausprobieren vorgeschlagen. Ich würde das noch weiter vereinfachen. Die verschachtelte Variable ist gar nicht nötig und folgendes ist besser lesbar:
`<description><de>${desc}</de></description>` und Variable `<desc><de>Der Makler versucht %suchende_person2% %wohnobjekt% %artofthedeal%.|%suchende_person% %wohnobjekt% %wohngrund%</de></desc>`
